pkg/upstream/cluster/loadbalancer.go

```diff
@@ -320,20 +336,97 @@ func (lb *EdfLoadBalancer) HostNum(metadata api.MetadataMatchCriteria) int {
 	return lb.hosts.Size()
 }
 
-func newEdfLoadBalancerLoadBalancer(hosts types.HostSet, unWeightChoose func(types.LoadBalancerContext) types.Host, hostWeightFunc func(host WeightItem) float64) *EdfLoadBalancer {
+func newEdfLoadBalancer(info types.ClusterInfo, hosts types.HostSet, unWeightChoose func(types.LoadBalancerContext) types.Host, hostWeightFunc func(host WeightItem) float64) *EdfLoadBalancer {
+	hostWeightFunc = slowStartHostWeightFunc(info, hostWeightFunc)
 	lb := &EdfLoadBalancer{
 		hosts:                  hosts,
 		rand:                   rand.New(rand.NewSource(time.Now().UnixNano())),
 		unweightChooseHostFunc: unWeightChoose,
 		hostWeightFunc:         hostWeightFunc,
 	}
-	lb.refresh(hosts)
+	lb.refresh(info, hosts)
 	return lb
 }
 
-func (lb *EdfLoadBalancer) refresh(hosts types.HostSet) {
-	// Check if the original host weights are equal and skip EDF creation if they are
-	if hostWeightsAreEqual(hosts) {
+func slowStartDurationFactorFunc(info types.ClusterInfo, host types.Host) float64 {
+	return slowStartDurationFactorFuncWithNowFunc(info, host, time.Now)
+}
+
+// slowStartDurationFactorFuncWithNowFunc with nowFunc parameter for testing
+func slowStartDurationFactorFuncWithNowFunc(info types.ClusterInfo, host types.Host, nowFunc func() time.Time) float64 {
+	slowStart := info.SlowStart()
+
+	if slowStart.SlowStartDuration <= 0 {
+		return 1.0
+	}
+
+	duration := nowFunc().Sub(host.LastHealthCheckPassTime())
```

Comment:
if the value is true of `lastHealthCheckPassTime.IsZero()`, we should return `1.0` before `Sub` operation.

Reply:
As above.

Reply:
As above, at most situations, `lastHealthCheckPassTime` should not be zero, so it's a unnecessary checking.